corporate/views/remote_billing_page.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from typing import Any, Dict, Optional, TypedDict
+
+from django.conf import settings
+from django.core import signing
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import render
+from django.urls import reverse
+from django.utils.translation import gettext as _
+from django.views.decorators.csrf import csrf_exempt
+
+from zerver.lib.exceptions import JsonableError, MissingRemoteRealmError
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.validator import check_bool, check_dict_only, check_float, check_int, check_string
+from zilencer.models import RemoteRealm, RemoteZulipServer
+
+billing_logger = logging.getLogger("corporate.stripe")
+
+
+SIGNED_ACCESS_TOKEN_VALIDITY_IN_SECONDS = 60 * 60 * 24
+
+
+class RemoteBillingIdentityDict(TypedDict):
+    user_uuid: str
+    user_email: str
+    user_full_name: str
+    remote_server_uuid: str
+    remote_realm_uuid: Optional[str]
+
+
+@csrf_exempt
+@has_request_variables
+def remote_server_billing_entry(
+    request: HttpRequest,
+    remote_server: RemoteZulipServer,
+    user_uuid: str = REQ(json_validator=check_string),
+    user_email: str = REQ(json_validator=check_string),
+    user_full_name: str = REQ(json_validator=check_string),
+    realm_info: Dict[str, Any] = REQ(
+        json_validator=check_dict_only(
+            required_keys=[
+                ("id", check_int),
+                ("host", check_string),
+                ("url", check_string),
+                ("deactivated", check_bool),
+                ("date_created", check_float),
+            ],
+            # Servers before 8.0 don't have these keys.
+            optional_keys=[
+                ("uuid", check_string),
+                ("uuid_owner_secret", check_string),
+            ],
+        )
+    ),
+) -> HttpResponse:
+    remote_realm = None
+    if "uuid" in realm_info:
+        # This codepath is for servers version 8.0+, as they submit the realm uuid,
+        # allowing us to tie this process to a RemoteRealm.
```

Comment:
I also considered whether the remote server should just send its `version` to the endpoint, and we'd make the decision based on that. But I think current approach is cleaner - a pre-8.0 server won't have `realm.uuid` fields, so it won't send them, and that's how this endpoint know it's dealing with <8.0.

Reply:
Yeah that seems OK. Also a pre-8.0 server won't have `realm_info` at all, probably.

Reply:
Ah okay, I thought we'd backport a limited version of this flow to 7.x or something, so it'd still be similar, just wouldn't send the uuid, but would send other stuff

Reply:
If we're backporting to 7.x, we might just backport a migration to create `realm.uuid` too.

Reply:
Hmm, okay, I cut out this check for now and have the flow assume we have `RemoteRealm`. It's a simple change to bring back in some shape if needed once we know exactly what we're doing with 7.x

Reply:
Sounds good.